src/libraries/System.Drawing.Common/src/System/Drawing/Icon.Windows.COMWrappers.cs

```diff
@@ -52,13 +54,25 @@ public unsafe void Save(Stream outputStream)
 
                     if (lpPicture != IntPtr.Zero)
                     {
-                        // don't assert the count went to 0 here because some other thread could be using the same lpPicture
-                        Marshal.Release(lpPicture);
+                        int count = Marshal.Release(lpPicture);
+                        Debug.Assert(count == 0);
                     }
+
                 }
             }
         }
 
+        private static void CheckSaveAsFileResult(int errorCode)
+        {
+            // Pass -1 for errorInfo to indicate that Windows' GetErrorInfo shouldn't be called, and only
+            // return the Exception corresponding to the specified errorCode.
+            Exception? ex = Marshal.GetExceptionForHR(errorCode, errorInfo: new IntPtr(-1));
+            if (ex != null)
+            {
+                throw ex;
+            }
```

Comment:
```suggestion
            Marshal.ThrowExceptionForHR(errorCode, errorInfo: new IntPtr(-1));
```

Reply:
I really need to learn all these Marshal APIs. 😉